src/NHibernate/Engine/Query/QueryPlanCache.cs

```diff
@@ -66,12 +67,14 @@ public IQueryExpressionPlan GetHQLQueryPlan(IQueryExpression queryExpression, bo
 				{
 					log.Debug("unable to locate HQL query plan in cache; generating ({0})", queryExpression.Key);
 				}
+				Stopwatch stopwatch = log.IsWarnEnabled() ? Stopwatch.StartNew() : null;
 				plan = new QueryExpressionPlan(queryExpression, shallow, enabledFilters, factory);
+				stopwatch?.Stop();
 				// 6.0 TODO: add "CanCachePlan { get; }" to IQueryExpression interface
 				if (!(queryExpression is ICacheableQueryExpression linqExpression) || linqExpression.CanCachePlan)
 					planCache.Put(key, plan);
 				else
-					log.Debug("Query plan not cacheable");
+					log.Warn("Query plan is not cacheable QueryKey:{0}, ElapsedMilliseconds:{1}", queryExpression.Key, stopwatch?.ElapsedMilliseconds);
```

Comment:
Please change to info.

Reply:
Let's not include this change at all. Is it even right place? If all the problems due Compile calls I'm not even sure this change includes them in stpwatch. If I remember correctly all Compile calls are already called at this stage.

Reply:
It's another issue mentioned [here](https://github.com/nhibernate/nhibernate-core/issues/2952#issuecomment-983636709),  if query is not cacheable at all developers must be warned/informed and change it if it's necessary. 
If you ok with that, i would like to change to information.

Reply:
Agree

Reply:
ok I revert commit and force push changes